### DISCUSSION TITLE
v1.29

### DISCUSSION THREAD

**Discussion by Joshua-Riek** [Joshua-Riek/ubuntu-rockchip#435] [Announcements]
Hello everyone! I want to share a few essential changes made in the recent release.

I created a few new commands to assist with flashing the bootloader and installing your current Ubuntu 22.04 system onto other media, such as an NVMe.

> For users with any version before v1.29, you can update your system with `sudo apt update && sudo apt upgrade -y` to install the U-Boot commands. 
>
> I have yet to package the new ubuntu-rockchip-install command, so it will only be found in the latest v1.29 release. I will package this in the future to issue improvements and bug fixes to the specific command.

## u-boot-install 
The command `u-boot-install` can update the bootloader on your SD Card or eMMC. 
```
sudo u-boot-install /dev/mmcblk0
```

## u-boot-install-mtd
The command `u-boot-install-mtd` can install the bootloader to the SPI Flash. 
```
sudo u-boot-install-mtd /dev/mtdblock0
```

## u-boot-install-sata-mtd
The command `u-boot-install-sata-mtd` can install the bootloader with SATA support to the SPI Flash. Right now, this command is only available on the Orange Pi 5.
```
sudo u-boot-install-sata-mtd /dev/mtdblock0
```

## ubuntu-rockchip-install
The command `ubuntu-rockchip-install` can copy your currently running system onto an NVMe, eMMC, or USB drive.
```
sudo ubuntu-rockchip-install /dev/mmcblk0
```

**Comment by JFLim1** [comment]
Hi @Joshua-Riek,

On Orange Pi 5 Plus.  I am interested in installing https://github.com/Joshua-Riek/ubuntu-rockchip/releases/download/v1.29/ubuntu-22.04.3-preinstalled-desktop-arm64-orangepi-5-plus.img.xz on SD card then use "sudo ubuntu-rockchip-install" to copy to "NVMe".  

The both the SPI and NVMe is new blank NO partition at all.  Question after boot up v1.29 on SD Card, do I jsut issue command:

sudo ubuntu-rockchip-install /dev/nvmeXXX?  Where nvmveXXX is the NVMe.  Or are there more steps to be done first?

**Reply by Joshua-Riek** [reply]
You will need to update the SPI first, this will take a few minutes. But after that, you can use the ubuntu-rockchip-install command on your NVMe, as shown below.
```
sudo u-boot-install-mtd /dev/mtdblock0
sudo ubuntu-rockchip-install /dev/nvme0n1
```

**Reply by BaconSoong** [reply]
how to update the SPI？use any command or something else.

**Comment by JFLim1** [comment]
@Joshua-Riek , Thank you for the guidance.  So there is no requirement to have any partition pre-setup on the NVMe or SPI?

**Reply by Joshua-Riek** [reply]
Correct, the commands will create the correct partitions automatically.

**Comment by emk2203** [comment]
What is the current u-boot version written by the `u-boot-install` command? I couldn't find this in the source or elsewhere.

Also, is kernel 6.6.0 useable with the Rock 5B? I saw that you bumped the version yesterday for mainline, but the Rock 5B 1.29 image is still running stock 5.10.160-rockchip.

**Reply by Joshua-Riek** [reply]
> What is the current u-boot version written by the u-boot-install command? I couldn't find this in the source or elsewhere.

The complete packaged source code for u-boot can be found on my PPA linked below. But all boards use the vendor u-boot v.2017 except for the Turing RK1, which uses v2024.01-rc1.

The u-boot version can also be seen in the UART logs during boot time.

https://launchpad.net/~jjriek/+archive/ubuntu/rockchip

> Also, is kernel 6.6.0 useable with the Rock 5B? I saw that you bumped the version yesterday for mainline, but the Rock 5B 1.29 image is still running stock 5.10.160-rockchip.

Yes, the 6.6.0 kernel is usable, but it does not have all the hardware features available, such as HDMI, GPU, and NPU. So, I promote the BSP kernel that uses 5.10.160 first, as all hardware will work as expected.

I prefer not to be bombarded with mainline Linux issues or questions, as it's a work in progress. There is a tagged 6.6.0 release that can be downloaded below.

https://github.com/Joshua-Riek/ubuntu-rockchip/releases/tag/v1.29.1

**Comment by Ukhupacha** [comment]
Hello @Joshua-Riek,

Do we need to `sudo u-boot-install-mtd /dev/mtdblock0 ` even if I dd the image directly to the nvme drive?

**Reply by Joshua-Riek** [reply]
Yes

**Comment by JFLim1** [comment]
Hi @Joshua-Riek,

Not sure whether I did some wrong or missing steps. Trying to install Ubuntu-22.04.3_v1.29 running on SD Card to NVMe.  

```
jfl@jfl-opi5plus:~$ lsblk
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
mtdblock0     31:0    0    16M  0 disk 
mmcblk0      179:0    0   233G  0 disk 
mmcblk0boot0 179:32   0     4M  1 disk 
mmcblk0boot1 179:64   0     4M  1 disk 
mmcblk1      179:96   0  59.5G  0 disk 
├─mmcblk1p1  179:97   0   512M  0 part /boot/firmware
└─mmcblk1p2  179:98   0    59G  0 part /
nvme0n1      259:0    0 476.9G  0 disk 
```
Supposed it managed to install the bootloader successfully .

```
jfl@jfl-opi5plus:~$ sudo u-boot-install-mtd /dev/mtdblock0
[sudo] password for jfl: 
Writing u-boot to the SPI flash
32768+0 records in
32768+0 records out
16777216 bytes (17 MB, 16 MiB) copied, 234.675 s, 71.5 kB/s
jfl@jfl-opi5plus:~$

```
But it somehow did not manage to install ubuntu-22.04.3_v1.29 onto NVMe.

```
jfl@jfl-opi5plus:~$ sudo ubuntu-rockchip-install /dev/nvme0n1
This script will install the currently running system onto /dev/nvme0n1.
It may take up to a few minutes - Continue? [y/N] y

Creating partition table for /dev/nvme0n1.
Creating filesystem on partitions /dev/nvme0n1p1 and /dev/nvme0n1p2.

blockdev: ioctl error on BLKRRPART: Device or resource busy
Error: in /usr/bin/ubuntu-rockchip-install on line 114
jfl@jfl-opi5plus:~$
```
OK, on second try "sudo ubuntu-rockchip-install /dev/nvme0n1" it managed to write NVMe.

```
jfl@jfl-opi5plus:~$ sudo ubuntu-rockchip-install /dev/nvme0n1
This script will install the currently running system onto /dev/nvme0n1.
It may take up to a few minutes - Continue? [y/N] y

Creating partition table for /dev/nvme0n1.
Creating filesystem on partitions /dev/nvme0n1p1 and /dev/nvme0n1p2.

Counting files...
Transferring 429 files (101 MB) from the bootfs to /dev/nvme0n1p1. Please wait!
Cleaning up...

Counting files...
Transferring 182789 files (7544 MB) from the rootfs to /dev/nvme0n1p2. Please wait!
Cleaning up...

Done!
jfl@jfl-opi5plus:~$
```
```
jfl@jfl-opi5plus:~$ lsblk
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
mtdblock0     31:0    0    16M  0 disk 
mmcblk0      179:0    0   233G  0 disk 
mmcblk0boot0 179:32   0     4M  1 disk 
mmcblk0boot1 179:64   0     4M  1 disk 
mmcblk1      179:96   0  59.5G  0 disk 
├─mmcblk1p1  179:97   0   512M  0 part /boot/firmware
└─mmcblk1p2  179:98   0    59G  0 part /
nvme0n1      259:0    0 476.9G  0 disk 
├─nvme0n1p1  259:5    0   512M  0 part 
└─nvme0n1p2  259:6    0 476.4G  0 part 
jfl@jfl-opi5plus:~$ 
```

**Reply by Joshua-Riek** [reply]
Weird. Try running the command `sync` then try to install it onto nvme again. If you run into problems, please try to reboot.

It looks like the nvme is busy.

**Comment by JFLim1** [comment]
> Weird. Try running the command `sync` then try to install it onto nvme again. If you run into problems, please try to reboot.

Ok on second try it seems to work without having to issue command "sync".  Will reboot to see whether it will boot successfully from NVMe.  Will revert later.

**Reply by Joshua-Riek** [reply]
Got it, thanks for reporting this. Since the installation script has only been released for a few days, it may contain a few bugs, I will try to do some research and improve where I can.

**Comment by JFLim1** [comment]
@Joshua-Riek,  Thank you for your great support.  Successfully booted from NVMe **without** SD Card image.

```
jfl@jfl-opi5plus:~$ lsblk
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda            8:0    1   3.6G  0 disk 
└─sda1         8:1    1   3.6G  0 part /media/jfl/EADAD935DAD8FEAF
mtdblock0     31:0    0    16M  0 disk 
mmcblk0      179:0    0   233G  0 disk 
mmcblk0boot0 179:32   0     4M  1 disk 
mmcblk0boot1 179:64   0     4M  1 disk 
nvme0n1      259:0    0 476.9G  0 disk 
├─nvme0n1p1  259:1    0   512M  0 part /boot/firmware
└─nvme0n1p2  259:2    0 476.4G  0 part /
jfl@jfl-opi5plus:~$ 
```
Have a question on the system boot up priority?  Is going to be:

1.  SD Card
2.  USB 
3.  NVMe
4.  EMMC

Or is based on different boot sequence?

**Reply by Joshua-Riek** [reply]
The boot order should be as below. (I forget if USB is before or after NVMe).

- SD Card
- USB
- NVMe
- eMMC

**Reply by emk2203** [reply]
This boot order would be the most sensible for repair purposes, I hope USB comes before NVMe.

**Comment by JFLim1** [comment]
Hi @Joshua-Riek,

Device: Orange Pi 5 Plus.  

After installing bootloader on SPL and Ubuntu-22.04.3_v1.29 on NVMe, am facing some issues  If boot up  with NVMe, the USB Drives (USB Flash Drive or USB HDD) are NO longer accessible: No permission to access the drive.

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/90e51213-eae2-486a-b89e-7c01a39be9ff)

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/b1068a40-13f0-465e-8539-c583b38ff6f7)


Any suggestion how to resolve this issue?

**Reply by Joshua-Riek** [reply]
What is the output of lsblk with your usb inserted? No changes where made with USB, so it should be working fine.

**Reply by JFLim1** [reply]
Basically I cannot access any USB Drives on any of the USB 2.0 or USB 3.0 Ports or USB 3.0 Hub.  "No permission to access/view the content of the drive" when boot up with NVMe. 

If boot up with SD Card, can access the same USB Drives.

```
jfl@jfl-opi5plus:~$ lsblk
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda            8:0    1   3.6G  0 disk 
└─sda1         8:1    1   3.6G  0 part /media/jfl/EADAD935DAD8FEAF
sdb            8:16   1   3.7G  0 disk 
└─sdb1         8:17   1   398M  0 part /media/jfl/MULTI_BOOT
mtdblock0     31:0    0    16M  0 disk 
mmcblk0      179:0    0   233G  0 disk 
mmcblk0boot0 179:32   0     4M  1 disk 
mmcblk0boot1 179:64   0     4M  1 disk 
nvme0n1      259:0    0 476.9G  0 disk 
├─nvme0n1p1  259:1    0   512M  0 part /boot/firmware
└─nvme0n1p2  259:2    0 476.4G  0 part /
jfl@jfl-opi5plus:~$
```

Boot with SD Card and the USB drives are accessible.

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/53d7d5db-b7f1-4251-a6a0-c164ac048745)

**Reply by Joshua-Riek** [reply]
Oh you need to take ownership of the device. Mount your usb to /mnt and run the below command as your current user on nvme. 
```
sudo chown -R $(whoami) /mnt 
```

**Reply by JFLim1** [reply]
Pardon my ignorance how to I mount usb to /mnt?  

I ran the command:

`sudo chown -R $(whoami) /mnt `

```
jfl@jfl-opi5plus:~$ sudo chown -R $(whoami) /mnt
[sudo] password for jfl:
```
Same issue cannot access to the USB Drives. I don't think I had mount the "usb to /mnt".

**Reply by JFLim1** [reply]
Tried this commands.

```
jfl@jfl-opi5plus:~$ sudo mount /dev/sda1 /mnt
jfl@jfl-opi5plus:~$ sudo mount /dev/sdb1 /mnt
jfl@jfl-opi5plus:~$ sudo mount /dev/sdc1 /mnt
```
 
```
jfl@jfl-opi5plus:~$ sudo chown -R $(whoami) /mnt
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-hugsun-x99.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3308-roc-cc.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-khadas-edge-v.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-nanopi-r4s.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399pro-rock-pi-n10.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-gru-scarlet-inx.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3368-lion-haikou.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3568-bpi-r2-pro.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3368-px5-evb.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-evb.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/px30-engicam-px30-core-edimm2.2.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-roc-pc-plus.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3568-rock-3a.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3328-roc-pc.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-pinebook-pro.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-sapphire-excavator.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-firefly.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-puma-haikou.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3326-odroid-go2.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-rock-pi-4b-plus.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-rock-pi-4a.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-kobol-helios64.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-khadas-edge-captain.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-gru-scarlet-kd.dtb': Operation not permitted
chown: changing ownership of '/mnt/dtbs/rockchip/rk3399-gru-kevin.dtb': Operation not permitted


chown: changing ownership of '/mnt': Operation not permitted
jfl@jfl-opi5plus:~$
```
Now the Nautilus no longer indicate that the 3 USB drives had been mounted.

**Reply by Joshua-Riek** [reply]
Try the below. Sorry im on mobile right now:

sudo chown -R jfl /mnt

**Reply by JFLim1** [reply]
```
jfl@jfl-opi5plus:~$ lsblk
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda            8:0    1   3.6G  0 disk 
└─sda1         8:1    1   3.6G  0 part 
mtdblock0     31:0    0    16M  0 disk 
mmcblk0      179:0    0   233G  0 disk 
mmcblk0boot0 179:32   0     4M  1 disk 
mmcblk0boot1 179:64   0     4M  1 disk 
nvme0n1      259:0    0 476.9G  0 disk 
├─nvme0n1p1  259:1    0   512M  0 part /boot/firmware
└─nvme0n1p2  259:2    0 476.4G  0 part /

jfl@jfl-opi5plus:~$ sudo mount /dev/sda1 /mnt
[sudo] password for jfl: 

jfl@jfl-opi5plus:~$ sudo chown -R jfl /mnt
chown: cannot access '/mnt/Aml_S922X_Armbian-Jammy-USB64GB': Input/output error
jfl@jfl-opi5plus:~$ 

jfl@jfl-opi5plus:~$ sudo chown -R $(whoami) /mnt
chown: cannot access '/mnt/Aml_S922X_Armbian-Jammy-USB64GB': Input/output error
jfl@jfl-opi5plus:~$
```

Using "gnome-disk-utility" after "sudo mount /dev/sda1 /mnt" it does seems to show it had mounted.  But I still have no access or permission to access the USB drive.

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/bd46a016-3dea-4c96-81b3-a8632511c599)

**Reply by JFLim1** [reply]
Have to call it a night for now.  2:32 am here.  Will continue tomorrow if it is ok with you. Thanks

**Reply by JFLim1** [reply]
Hi @Joshua-Riek, if nobody else is facing this issue when boot from NVMe, I try reinstall Ubuntu-22.04.3_v1.29 on NVMe again.

**Reply by Joshua-Riek** [reply]
This should be fixed, fat partitions can not be changed ownership. I think NTFS are the same, you must re format.

**Reply by JFLim1** [reply]
Now this is puzzling.  The original SD Card Image of Ubuntu-22.04.3_v1.29 can access all the USB drive I have be it EXT4, FAT or NTFS format.  But once installed on NVMe now it complain I don't have permission to access/view these USB drives and cannot change ownership.  Pardon my ignorance, why is it necessary to change ownership when boot with NVMe and not when boot up with SD Card?

Is there a reason why when boot up with SD Card all these USE drives are accessible but not when it is installed on NVMe SSD and boot up with NVMe SSD?  Isn't both image on SD Card and NVMe SSD should be the same?

I have reinstall v1.29 on the NVMe SSD again with the command:
```
sudo ubuntu-rockchip-install /dev/nvme0n1
sudo sync
```
The result is the same, No permission to access/view any USB drives.

**Reply by JFLim1** [reply]
@Joshua-Riek, had a similar issue with BredOS where the USB Stick and USB HDD with NTFS cannot be mounted (slight difference from Ubuntu-22.04.3_v1.29 on NVMe SSD where it cannot access/view the file) the outcome is the same though i.e. cannot use these NTFS formatted USB drives.

Fortunately on BredOS thee is a solution pointed out by @dyingbred by installing a package "ntfs-3g".  With "ntfs-3p" installed, now BredOS had no issue recognizing and access the NTFS formatted USB drives.

Wonder there there is a similar package in Ubuntu?  With NVMe SSD as the boot up drive but not being able to access or use external USB drives is limiting.

**Reply by Joshua-Riek** [reply]
You can install the same package on ubuntu.

**Reply by JFLim1** [reply]
Hi @Joshua-Riek,
Yes, "ntfs-3g" is available on Ubuntu and is preinstalled on Ubuntu-22.04.3_v1.29 (SD Card. I supposed on NVMe SSD image too, will confirm later).

On NVMe SSD "ntfs-3g" is also installed.
```
jfl@jfl-opi5plus:~$ apt list ntfs-3g
Listing... Done
ntfs-3g/jammy-updates,jammy-security,now 1:2021.8.22-3ubuntu1.2 arm64 [installed]
N: There is 1 additional version. Please use the '-a' switch to see it
jfl@jfl-opi5plus:~$
``` 
So "ntfs-3g" is not the solution.  It is weird why when boot up with NVMe SSD the system do not give permission to access/view any USB drives not just NTFS formatted drives. Boot up with SD Card all the USB Drives are accessible!

Question: Do you or any other OPI5 Plus users with NVMe SSD having the same or similar issue as I am encountering?

**Reply by JFLim1** [reply]
Hi @Joshua-Riek,

Kind of stumble on a partial solution of sort to resolve the issue when boot up with NVMe SSD and not having the permission to view the context of any USB drives as highlighted above.

If start Nautilsu using **"sudo nautilus"** and enter the password it can access:
`sudo nautilus`

NTFS Formatted USB Flash Drive
![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/5fb8aae8-05bf-4128-8eed-2f8440f88dbc)

EXT4 Formatted USB Flash Drive
![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/75384411/5ad0f01a-ba78-4fe7-aebe-9833b32aafb9)

I am a greenhorn in Linux.  Is there a better way to resolve this? 

Each application need to launch as "root" in order to access the USB drives.  Using GUI to launch does NOT not render the app to be able access the USB drives.

Does it looks like when Ubuntu-22.04.3_v1.29 was installed on NVMe SSD, the user "jfl" was given less access permission.  Is there a way to restore the user "jfl" having all or similar to "root" or something like that?

**Reply by JFLim1** [reply]
Hi @Joshua-Riek,

Would you think these command to change owner is safe?

sudo chown $USER /media/$USER

sudo chown jfl /media/jfl

or 

cd /media
sudo setfacl -m u:$USER:rwx $USER

**Reply by Joshua-Riek** [reply]
@JFLim1 Please open a separate issue, this is borderline spam.

**Comment by JFLim1** [comment]
Hi @Joshua-Riek,

OPI 5 Plus.  With the bootloader installed on SPL and Ubuntu-22.04.3_v1.29 installed on NVMe SSD when tried to boot from USB Flash (with mainline kernel) but OPI 5 Plus boot up NVMe SSD Ubuntu-22.04.3_v1.29 instead of with USB.

Is there a way to set the priority of the boot sequence from the storage devices?  Example SD->USB->NVMe->EMMC? or USB->SD->NVMe->EMMC?

**Reply by Joshua-Riek** [reply]
The bootloader is a terrible hack from rockchip, and I would prefer not to go digging through it. I'm not even sure if all the usb ports are defined in the device tree and work.

**Reply by JFLim1** [reply]
Thanks for the info.  It is beyond my skill set to dig through the bootloader.  

Just thought that SD->USB->NVMe->EMMC or USB->SD->NVMe->EMMC sound logical booting priority as both USB drive and SD Card are external drives which should be given priority in case the EMMC or NVMe have issue.

**Comment by theflyingape** [comment]
To clarify, NIC was working originally as I was able to do an apt update / apt upgrade.  After this flash / copy:
```
sudo u-boot-install-mtd /dev/mtdblock0
sudo ubuntu-rockchip-install /dev/nvme0n1
```
... but the issue posted here about the **missing ethernet interface** is the same I am seeing: https://github.com/MichaIng/DietPi/issues/6663
Even after pulling the NVMe and booting off the SD card again, I see a different MAC hardware address.

**Reply by Joshua-Riek** [reply]
I don't follow, I fixed the mac address issue. Are you saying there is no ethernet interface?

**Reply by theflyingape** [reply]
Correct, no NIC.
After booting SD card, NetworkManager shows me what /sys/class/net/eth0/address has: `82:78:C6:83:CA:19` but any attempts to use eth0 (ifconfig eth0 up, dhclient eth0) fails with 'No such device'.
When booting off NVMe, it shows eth0 as `ae:ab:a4:65:42:b4` and no joy lighting the interface up.
I just re-imaged the SD card from this image download, booted, and it shows yet another MAC `a6:03:11:0e:e6:df` but no device to attach an IP.
So, I re-ran your two steps and re-booted (warm first, then cold) and got different MAC addresses each time.

**Reply by Joshua-Riek** [reply]
Sorry, im so confused. You dont have a network interface but you have a mac address? I dont see how that is possible.

**Reply by theflyingape** [reply]
Yes, it's behaving like the recent issue I posted, whereas it's randomly assigning a new MAC hardware address -- I can only guess when devices are enumerated, it captures the address at that point-in-time, but then somehow it's getting assigned a new address later on causing a disconnect with the userspace controls?

This was not happening before, until after that flash. I'll go back to an earlier image and see if I can negate this "feature".

**Reply by theflyingape** [reply]
Attempted to use v1.21 with #274 fix, re-flash, but to no avail.  Re-booted under v1.29 and gathered some log info if it helps.
Perhaps the NIC itself got "fried" as it shows the hardware attached but "disabled"?
[rk_gmac-dwmac.log](https://github.com/Joshua-Riek/ubuntu-rockchip/files/13246066/rk_gmac-dwmac.log)

**Comment by acruvium** [comment]
I have a orange pi 5 plus with 16GB memory and installed nvme ssd. a few days ago, I installed 1.29 version and ssh connection lost about 20 minutes later. especially I did not install any program.

**Reply by Joshua-Riek** [reply]
I need more information to know the cause of this. It could be your power supply or network.

**Reply by acruvium** [reply]
I use samsung 25w PD charger(C to C) and wired network(ASUS GT-AXE11000).

**Comment by Not-a-good-username** [comment]
I have an orange pi 5 plus and flash the os to the tfcard, then I use dd to transfer os to emmc, and it boot from emmc correctly when board has no tfcard. but when I plug an empty tfcard（only a ntfs partition）, the board just can't boot. i want to boot from emmc with a tfcard (no system, only for storage) , how should I do?

**Reply by emk2203** [reply]
You can not boot the machine remote, or can you do this?

Then, the answer is simple: Insert the TFCard *after* you boot.

**Reply by Joshua-Riek** [reply]
I think this is the problem, it also provides a fix.

https://github.com/orangepi-xunlong/u-boot-orangepi/pull/11

**Comment by Not-a-good-username** [comment]
> You can not boot the machine remote, or can you do this?
> 
> Then, the answer is simple: Insert the TFCard _after_ you boot.

that‘s not elegant！(laugh)
but if there is no solution, i'll try that.

**Comment by Not-a-good-username** [comment]
> I think this is the problem, it also provides a fix.
> 
> [orangepi-xunlong/u-boot-orangepi#11](https://github.com/orangepi-xunlong/u-boot-orangepi/pull/11)

i'll try that, thanks.

**Reply by Joshua-Riek** [reply]
When I get home after work, I can apply the patch and push an update to the bootloader. Will take me a few hours from now.

**Reply by ebw44** [reply]
@Joshua-Riek 
I tried to update my bootloader on the emmc of my orange pi 5b to fix the https://github.com/orangepi-xunlong/u-boot-orangepi/pull/11
I use `sudo u-boot-install /dev/mmcblk0` both from my current emmc install (after `apt upgrade`) and from an SD card with the latest version ([v1.33](https://github.com/Joshua-Riek/ubuntu-rockchip/releases/tag/v1.33)). But I still have the issue of not booting (on the emmc) if there is a non-bootable SD card.

I saw that the [uboot patch was applied](https://github.com/Joshua-Riek/ubuntu-rockchip/commit/5391faf4868bc45c57defdfb316464e03cf1b8fd) but I'm not sure if `u-boot-install` uses a version of uboot that has the patch. Do I need to force a download of the latest version of uboot?

Thanks for the great work.

**Comment by Xeue** [comment]
You mentioned elsewhere you had an issue with hosting the image files as they are too large? Do you have any plan for this atm? If there's a way I can help, let me know, I can probably host something that can facilitate this if needed.
I primarily do webdev, so more than happy to help with hosting/building a page/site for it if that is of any use?

**Comment by danielmerja** [comment]
will the u-boot-install-mtd command format the SPI as well or just install the bootloader firmware onto the designated SPI?
